In [151]:
import pandas as pd
pd.set_option('display.max_columns', None)

df = pd.read_csv("../../main/h&m_dataset/articles_hm.csv")
df.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  colour_group_code colour_group_name  \
0                     Solid                  9             Black   
1                     Solid                 10             White   
2                    Stripe                 11         Off White   
3                     Solid                  9             Black   
4                     Solid                 10             White   

   perceived_colour_value_id perceived_colour_value_name  \
0                          4                        Dark   
1                          3                       Light   
2                          1                 Dusty Light   
3                          4                        Dark   
4                          3                       Light   

   perceived_colour_master_id perceived_colour_master_name  department_no  \
0                           5                        Black           1676   
1                           9                        White           1676   
2                           9                        White           1676   
3                           5                        Black           1339   
4                           9                        White           1339   

  department_name index_code        index_name  index_group_no  \
0    Jersey Basic          A        Ladieswear               1   
1    Jersey Basic          A        Ladieswear               1   
2    Jersey Basic          A        Ladieswear               1   
3  Clean Lingerie          B  Lingeries/Tights               1   
4  Clean Lingerie          B  Lingeries/Tights               1   

  index_group_name  section_no            section_name  garment_group_no  \
0       Ladieswear          16  Womens Everyday Basics              1002   
1       Ladieswear          16  Womens Everyday Basics              1002   
2       Ladieswear          16  Womens Everyday Basics              1002   
3       Ladieswear          61         Womens Lingerie              1017   
4       Ladieswear          61         Womens Lingerie              1017   

  garment_group_name                                        detail_desc  
0       Jersey Basic            Jersey top with narrow shoulder straps.  
1       Jersey Basic            Jersey top with narrow shoulder straps.  
2       Jersey Basic            Jersey top with narrow shoulder straps.  
3  Under-, Nightwear  Microfibre T-shirt bra with underwired, moulde...  
4  Under-, Nightwear  Microfibre T-shirt bra with underwired, moulde...

In [152]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 105542 entries, 0 to 105541
Data columns (total 25 columns):
 #   Column                        Non-Null Count   Dtype
---  ------                        --------------   -----
 0   article_id                    105542 non-null  int64
 1   product_code                  105542 non-null  int64
 2   prod_name                     105542 non-null  str  
 3   product_type_no               105542 non-null  int64
 4   product_type_name             105542 non-null  str  
 5   product_group_name            105542 non-null  str  
 6   graphical_appearance_no       105542 non-null  int64
 7   graphical_appearance_name     105542 non-null  str  
 8   colour_group_code             105542 non-null  int64
 9   colour_group_name             105542 non-null  str  
 10  perceived_colour_value_id     105542 non-null  int64
 11  perceived_colour_value_name   105542 non-null  str  
 12  perceived_colour_master_id    105542 non-null  int64
 13  perceived_colour_master_n

In [153]:
df.shape

(105542, 25)

In [154]:
df.isna().sum()

article_id                        0
product_code                      0
prod_name                         0
product_type_no                   0
product_type_name                 0
product_group_name                0
graphical_appearance_no           0
graphical_appearance_name         0
colour_group_code                 0
colour_group_name                 0
perceived_colour_value_id         0
perceived_colour_value_name       0
perceived_colour_master_id        0
perceived_colour_master_name      0
department_no                     0
department_name                   0
index_code                        0
index_name                        0
index_group_no                    0
index_group_name                  0
section_no                        0
section_name                      0
garment_group_no                  0
garment_group_name                0
detail_desc                     416
dtype: int64

총 데이터 개수 : 105542  
결측치 : detail_desc 416건 → 상품 설명인데 굳이 필요할까? 제거해도 분석 자체에는 문제 없을 듯  

### detail_desc 컬럼 자체 제거

In [155]:
df_drop = df.drop('detail_desc', axis=1)
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  colour_group_code colour_group_name  \
0                     Solid                  9             Black   
1                     Solid                 10             White   
2                    Stripe                 11         Off White   
3                     Solid                  9             Black   
4                     Solid                 10             White   

   perceived_colour_value_id perceived_colour_value_name  \
0                          4                        Dark   
1                          3                       Light   
2                          1                 Dusty Light   
3                          4                        Dark   
4                          3                       Light   

   perceived_colour_master_id perceived_colour_master_name  department_no  \
0                           5                        Black           1676   
1                           9                        White           1676   
2                           9                        White           1676   
3                           5                        Black           1339   
4                           9                        White           1339   

  department_name index_code        index_name  index_group_no  \
0    Jersey Basic          A        Ladieswear               1   
1    Jersey Basic          A        Ladieswear               1   
2    Jersey Basic          A        Ladieswear               1   
3  Clean Lingerie          B  Lingeries/Tights               1   
4  Clean Lingerie          B  Lingeries/Tights               1   

  index_group_name  section_no            section_name  garment_group_no  \
0       Ladieswear          16  Womens Everyday Basics              1002   
1       Ladieswear          16  Womens Everyday Basics              1002   
2       Ladieswear          16  Womens Everyday Basics              1002   
3       Ladieswear          61         Womens Lingerie              1017   
4       Ladieswear          61         Womens Lingerie              1017   

  garment_group_name  
0       Jersey Basic  
1       Jersey Basic  
2       Jersey Basic  
3  Under-, Nightwear  
4  Under-, Nightwear

In [156]:
df_drop.info()

<class 'pandas.DataFrame'>
RangeIndex: 105542 entries, 0 to 105541
Data columns (total 24 columns):
 #   Column                        Non-Null Count   Dtype
---  ------                        --------------   -----
 0   article_id                    105542 non-null  int64
 1   product_code                  105542 non-null  int64
 2   prod_name                     105542 non-null  str  
 3   product_type_no               105542 non-null  int64
 4   product_type_name             105542 non-null  str  
 5   product_group_name            105542 non-null  str  
 6   graphical_appearance_no       105542 non-null  int64
 7   graphical_appearance_name     105542 non-null  str  
 8   colour_group_code             105542 non-null  int64
 9   colour_group_name             105542 non-null  str  
 10  perceived_colour_value_id     105542 non-null  int64
 11  perceived_colour_value_name   105542 non-null  str  
 12  perceived_colour_master_id    105542 non-null  int64
 13  perceived_colour_master_n

In [157]:
df_drop.isna().sum()

article_id                      0
product_code                    0
prod_name                       0
product_type_no                 0
product_type_name               0
product_group_name              0
graphical_appearance_no         0
graphical_appearance_name       0
colour_group_code               0
colour_group_name               0
perceived_colour_value_id       0
perceived_colour_value_name     0
perceived_colour_master_id      0
perceived_colour_master_name    0
department_no                   0
department_name                 0
index_code                      0
index_name                      0
index_group_no                  0
index_group_name                0
section_no                      0
section_name                    0
garment_group_no                0
garment_group_name              0
dtype: int64

결측치는 우선 제거함

### 제품 색상 코드 컬럼 정리

제품 색상 코드를 나타내는 컬럼이 'colour_group_code', 'colour_group_name', 'perceived_colour_value_id', 'perceived_colour_value_name', 'perceived_colour_master_id', 'perceived_colour_master_name'  
이렇게 총 6개나 된다  
데이터를 단순화 하기 위해 'perceived_colour_master_id', 'perceived_colour_master_name' 2개 컬럼을 대표로 사용하고 나머지 컬럼은 삭제한다

In [ ]:
df_drop = df_drop.drop(['colour_group_code', 'colour_group_name', 'perceived_colour_value_id', 'perceived_colour_value_name'], axis=1)
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  
0  Womens Everyday Basics              1002       Jersey Basic  
1  Womens Everyday Basics              1002       Jersey Basic  
2  Womens Everyday Basics              1002       Jersey Basic  
3         Womens Lingerie              1017  Under-, Nightwear  
4         Womens Lingerie              1017  Under-, Nightwear

In [159]:
df_drop.info()

<class 'pandas.DataFrame'>
RangeIndex: 105542 entries, 0 to 105541
Data columns (total 20 columns):
 #   Column                        Non-Null Count   Dtype
---  ------                        --------------   -----
 0   article_id                    105542 non-null  int64
 1   product_code                  105542 non-null  int64
 2   prod_name                     105542 non-null  str  
 3   product_type_no               105542 non-null  int64
 4   product_type_name             105542 non-null  str  
 5   product_group_name            105542 non-null  str  
 6   graphical_appearance_no       105542 non-null  int64
 7   graphical_appearance_name     105542 non-null  str  
 8   perceived_colour_master_id    105542 non-null  int64
 9   perceived_colour_master_name  105542 non-null  str  
 10  department_no                 105542 non-null  int64
 11  department_name               105542 non-null  str  
 12  index_code                    105542 non-null  str  
 13  index_name               

In [160]:
df_drop.shape

(105542, 20)

**컬럼 수: 25개 → 22개**

In [161]:
df_drop.isna().sum()

article_id                      0
product_code                    0
prod_name                       0
product_type_no                 0
product_type_name               0
product_group_name              0
graphical_appearance_no         0
graphical_appearance_name       0
perceived_colour_master_id      0
perceived_colour_master_name    0
department_no                   0
department_name                 0
index_code                      0
index_name                      0
index_group_no                  0
index_group_name                0
section_no                      0
section_name                    0
garment_group_no                0
garment_group_name              0
dtype: int64

### 컬럼별 고유값 목록 확인

In [162]:
df_drop[['product_type_no', 'product_type_name']].value_counts()

product_type_no  product_type_name
272              Trousers             11169
265              Dress                10362
252              Sweater               9302
255              T-shirt               7904
254              Top                   4155
                                      ...  
366              Towel                    1
493              Wood balls               1
464              Pre-walkers              1
468              Bumbag                   1
483              Clothing mist            1
Name: count, Length: 132, dtype: int64

In [163]:
check = df_drop.groupby('product_type_no')['product_type_name'].nunique()
print(check[check > 1])

Series([], Name: product_type_name, dtype: int64)


product_type_no == product_type_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [164]:
df_drop['product_type_info'] = df_drop['product_type_name'] + " (" + df_drop['product_type_no'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info  
0    Vest top (253)  
1    Vest top (253)  
2    Vest top (253)  
3         Bra (306)  
4         Bra (306)

In [165]:
df_drop[['graphical_appearance_no', 'graphical_appearance_name']].value_counts()

graphical_appearance_no  graphical_appearance_name
 1010016                 Solid                        49747
 1010001                 All over pattern             17165
 1010010                 Melange                       5938
 1010017                 Stripe                        4990
 1010023                 Denim                         4842
 1010008                 Front print                   3215
 1010014                 Placement print               3098
 1010004                 Check                         2178
 1010005                 Colour blocking               1830
 1010021                 Lace                          1513
 1010026                 Other structure               1502
 1010002                 Application/3D                1341
 1010007                 Embroidery                    1165
 1010012                 Mixed solid/pattern           1132
 1010009                 Glittering/Metallic            958
 1010022                 Jacquard                

In [166]:
check2 = df_drop.groupby('graphical_appearance_no')['graphical_appearance_name'].nunique()
print(check2[check2 > 1])

Series([], Name: graphical_appearance_name, dtype: int64)


graphical_appearance_no == graphical_appearance_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [167]:
df_drop['graphical_appearance_info'] = df_drop['graphical_appearance_name'] + " (" + df_drop['graphical_appearance_no'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info  
0    Vest top (253)           Solid (1010016)  
1    Vest top (253)           Solid (1010016)  
2    Vest top (253)          Stripe (1010017)  
3         Bra (306)           Solid (1010016)  
4         Bra (306)           Solid (1010016)

In [168]:
df_drop[['perceived_colour_master_id', 'perceived_colour_master_name']].value_counts()

perceived_colour_master_id  perceived_colour_master_name
 5                          Black                           22585
 2                          Blue                            18469
 9                          White                           12665
 4                          Pink                             9403
 12                         Grey                             8924
 18                         Red                              5878
 11                         Beige                            5657
 19                         Green                            3526
 20                         Khaki green                      3181
 8                          Yellow                           3121
 3                          Orange                           2734
 13                         Brown                            2269
 15                         Metal                            2180
 7                          Turquoise                        1829
 1                 

In [169]:
check3 = df_drop.groupby('perceived_colour_master_id')['perceived_colour_master_name'].nunique()
print(check3[check3 > 1])

Series([], Name: perceived_colour_master_name, dtype: int64)


perceived_colour_master_id == perceived_colour_master_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [170]:
df_drop['perceived_colour_master_info'] = df_drop['perceived_colour_master_name'] + " (" + df_drop['perceived_colour_master_id'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info perceived_colour_master_info  
0    Vest top (253)           Solid (1010016)                    Black (5)  
1    Vest top (253)           Solid (1010016)                    White (9)  
2    Vest top (253)          Stripe (1010017)                    White (9)  
3         Bra (306)           Solid (1010016)                    Black (5)  
4         Bra (306)           Solid (1010016)                    White (9)

In [171]:
df_drop[['department_no', 'department_name']].value_counts()

department_no  department_name                
7616           Kids Girl Jersey Fancy             2032
1338           Expressive Lingerie                1921
8716           Young Girl Jersey Fancy            1874
4242           Swimwear                           1839
7648           Kids Boy Jersey Fancy              1488
                                                  ... 
5787           Shirt Extended inactive from s1       1
7659           Kids Boy License                      1
1620           Jersey inactive from S.6              1
3943           EQ Divided Blue                       1
3931           Accessories Other                     1
Name: count, Length: 299, dtype: int64

In [172]:
check4 = df_drop.groupby('department_no')['department_name'].nunique()
print(check4[check4 > 1])

Series([], Name: department_name, dtype: int64)


department_no == department_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [173]:
df_drop['department_info'] = df_drop['department_name'] + " (" + df_drop['department_no'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info perceived_colour_master_info  \
0    Vest top (253)           Solid (1010016)                    Black (5)   
1    Vest top (253)           Solid (1010016)                    White (9)   
2    Vest top (253)          Stripe (1010017)                    White (9)   
3         Bra (306)           Solid (1010016)                    Black (5)   
4         Bra (306)           Solid (1010016)                    White (9)   

         department_info  
0    Jersey Basic (1676)  
1    Jersey Basic (1676)  
2    Jersey Basic (1676)  
3  Clean Lingerie (1339)  
4  Clean Lingerie (1339)

In [174]:
df_drop[['index_code', 'index_name']].value_counts()

index_code  index_name                    
A           Ladieswear                        26001
D           Divided                           15149
F           Menswear                          12553
H           Children Sizes 92-140             12007
I           Children Sizes 134-170             9214
G           Baby Sizes 50-98                   8875
C           Ladies Accessories                 6961
B           Lingeries/Tights                   6775
J           Children Accessories, Swimwear     4615
S           Sport                              3392
Name: count, dtype: int64

In [175]:
check5 = df_drop.groupby('index_code')['index_name'].nunique()
print(check5[check5 > 1])

Series([], Name: index_name, dtype: int64)


index_code == index_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [176]:
df_drop['index_info'] = df_drop['index_name'] + " (" + df_drop['index_code'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info perceived_colour_master_info  \
0    Vest top (253)           Solid (1010016)                    Black (5)   
1    Vest top (253)           Solid (1010016)                    White (9)   
2    Vest top (253)          Stripe (1010017)                    White (9)   
3         Bra (306)           Solid (1010016)                    Black (5)   
4         Bra (306)           Solid (1010016)                    White (9)   

         department_info            index_info  
0    Jersey Basic (1676)        Ladieswear (A)  
1    Jersey Basic (1676)        Ladieswear (A)  
2    Jersey Basic (1676)        Ladieswear (A)  
3  Clean Lingerie (1339)  Lingeries/Tights (B)  
4  Clean Lingerie (1339)  Lingeries/Tights (B)

In [177]:
df_drop[['index_group_no', 'index_group_name']].value_counts()

index_group_no  index_group_name
1               Ladieswear          39737
4               Baby/Children       34711
2               Divided             15149
3               Menswear            12553
26              Sport                3392
Name: count, dtype: int64

In [178]:
check5 = df_drop.groupby('index_group_no')['index_group_name'].nunique()
print(check5[check5 > 1])

Series([], Name: index_group_name, dtype: int64)


index_group_no == index_group_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [179]:
df_drop['index_group_info'] = df_drop['index_group_name'] + " (" + df_drop['index_group_no'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info perceived_colour_master_info  \
0    Vest top (253)           Solid (1010016)                    Black (5)   
1    Vest top (253)           Solid (1010016)                    White (9)   
2    Vest top (253)          Stripe (1010017)                    White (9)   
3         Bra (306)           Solid (1010016)                    Black (5)   
4         Bra (306)           Solid (1010016)                    White (9)   

         department_info            index_info index_group_info  
0    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)  
1    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)  
2    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)  
3  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)  
4  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)

In [180]:
df_drop[['section_no', 'section_name']].value_counts()

section_no  section_name                  
15          Womens Everyday Collection        7295
53          Divided Collection                7124
44          Baby Essentials & Complements     4932
76          Kids Girl                         4469
77          Young Girl                        3899
61          Womens Lingerie                   3598
79          Girls Underwear & Basics          3490
11          Womens Tailoring                  3376
46          Kids Boy                          3328
66          Womens Small accessories          3270
6           Womens Casual                     2725
45          Kids Outerwear                    2665
18          Womens Trend                      2622
50          Divided Projects                  2364
47          Young Boy                         2352
2           H&M+                              2337
26          Men Underwear                     2322
8           Mama                              2266
42          Kids & Baby Shoes          

In [181]:
check6 = df_drop.groupby('section_no')['section_name'].nunique()
print(check6[check6 > 1])

Series([], Name: section_name, dtype: int64)


section_no == section_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [182]:
df_drop['section_info'] = df_drop['section_name'] + " (" + df_drop['section_no'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info perceived_colour_master_info  \
0    Vest top (253)           Solid (1010016)                    Black (5)   
1    Vest top (253)           Solid (1010016)                    White (9)   
2    Vest top (253)          Stripe (1010017)                    White (9)   
3         Bra (306)           Solid (1010016)                    Black (5)   
4         Bra (306)           Solid (1010016)                    White (9)   

         department_info            index_info index_group_info  \
0    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
1    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
2    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
3  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   
4  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   

                  section_info  
0  Womens Everyday Basics (16)  
1  Womens Everyday Basics (16)  
2  Womens Everyday Basics (16)  
3         Womens Lingerie (61)  
4         Womens Lingerie (61)

In [183]:
df_drop[['garment_group_no', 'garment_group_name']].value_counts()

garment_group_no  garment_group_name           
1005              Jersey Fancy                     21445
1019              Accessories                      11519
1002              Jersey Basic                      8126
1003              Knitwear                          7490
1017              Under-, Nightwear                 7441
1009              Trousers                          6727
1010              Blouses                           5838
1020              Shoes                             5145
1013              Dresses Ladies                    4874
1007              Outdoor                           4501
1001              Unknown                           3873
1016              Trousers Denim                    3100
1018              Swimwear                          2787
1021              Socks and Tights                  2272
1011              Shirts                            2116
1006              Woven/Jersey/Knitted mix Baby     1965
1025              Shorts                

In [184]:
check7 = df_drop.groupby('garment_group_no')['garment_group_name'].nunique()
print(check7[check7 > 1])

Series([], Name: garment_group_name, dtype: int64)


garment_group_no == garment_group_name 깔끔하게 서로 매칭됨 → 두 컬럼을 하나로 합치는 파생 컬럼 생성

In [185]:
df_drop['garment_group_info'] = df_drop['garment_group_name'] + " (" + df_drop['garment_group_no'].astype(str) + ")"
df_drop.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  perceived_colour_master_id  \
0                     Solid                           5   
1                     Solid                           9   
2                    Stripe                           9   
3                     Solid                           5   
4                     Solid                           9   

  perceived_colour_master_name  department_no department_name index_code  \
0                        Black           1676    Jersey Basic          A   
1                        White           1676    Jersey Basic          A   
2                        White           1676    Jersey Basic          A   
3                        Black           1339  Clean Lingerie          B   
4                        White           1339  Clean Lingerie          B   

         index_name  index_group_no index_group_name  section_no  \
0        Ladieswear               1       Ladieswear          16   
1        Ladieswear               1       Ladieswear          16   
2        Ladieswear               1       Ladieswear          16   
3  Lingeries/Tights               1       Ladieswear          61   
4  Lingeries/Tights               1       Ladieswear          61   

             section_name  garment_group_no garment_group_name  \
0  Womens Everyday Basics              1002       Jersey Basic   
1  Womens Everyday Basics              1002       Jersey Basic   
2  Womens Everyday Basics              1002       Jersey Basic   
3         Womens Lingerie              1017  Under-, Nightwear   
4         Womens Lingerie              1017  Under-, Nightwear   

  product_type_info graphical_appearance_info perceived_colour_master_info  \
0    Vest top (253)           Solid (1010016)                    Black (5)   
1    Vest top (253)           Solid (1010016)                    White (9)   
2    Vest top (253)          Stripe (1010017)                    White (9)   
3         Bra (306)           Solid (1010016)                    Black (5)   
4         Bra (306)           Solid (1010016)                    White (9)   

         department_info            index_info index_group_info  \
0    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
1    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
2    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
3  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   
4  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   

                  section_info        garment_group_info  
0  Womens Everyday Basics (16)       Jersey Basic (1002)  
1  Womens Everyday Basics (16)       Jersey Basic (1002)  
2  Womens Everyday Basics (16)       Jersey Basic (1002)  
3         Womens Lingerie (61)  Under-, Nightwear (1017)  
4         Womens Lingerie (61)  Under-, Nightwear (1017)

### 요약 컬럼 활용해서 df 재구성

In [187]:
new_df = df_drop[['article_id', 'product_code', 'prod_name', 'product_type_info', 'product_group_name', 'graphical_appearance_info', 'perceived_colour_master_info', 'department_info',	'index_info', 'index_group_info', 'section_info', 'garment_group_info']]
new_df.head()

article_id  product_code          prod_name product_type_info  \
0   108775015        108775          Strap top    Vest top (253)   
1   108775044        108775          Strap top    Vest top (253)   
2   108775051        108775      Strap top (1)    Vest top (253)   
3   110065001        110065  OP T-shirt (Idro)         Bra (306)   
4   110065002        110065  OP T-shirt (Idro)         Bra (306)   

   product_group_name graphical_appearance_info perceived_colour_master_info  \
0  Garment Upper body           Solid (1010016)                    Black (5)   
1  Garment Upper body           Solid (1010016)                    White (9)   
2  Garment Upper body          Stripe (1010017)                    White (9)   
3           Underwear           Solid (1010016)                    Black (5)   
4           Underwear           Solid (1010016)                    White (9)   

         department_info            index_info index_group_info  \
0    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
1    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
2    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
3  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   
4  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   

                  section_info        garment_group_info  
0  Womens Everyday Basics (16)       Jersey Basic (1002)  
1  Womens Everyday Basics (16)       Jersey Basic (1002)  
2  Womens Everyday Basics (16)       Jersey Basic (1002)  
3         Womens Lingerie (61)  Under-, Nightwear (1017)  
4         Womens Lingerie (61)  Under-, Nightwear (1017)

In [188]:
new_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 105542 entries, 0 to 105541
Data columns (total 12 columns):
 #   Column                        Non-Null Count   Dtype
---  ------                        --------------   -----
 0   article_id                    105542 non-null  int64
 1   product_code                  105542 non-null  int64
 2   prod_name                     105542 non-null  str  
 3   product_type_info             105542 non-null  str  
 4   product_group_name            105542 non-null  str  
 5   graphical_appearance_info     105542 non-null  str  
 6   perceived_colour_master_info  105542 non-null  str  
 7   department_info               105542 non-null  str  
 8   index_info                    105542 non-null  str  
 9   index_group_info              105542 non-null  str  
 10  section_info                  105542 non-null  str  
 11  garment_group_info            105542 non-null  str  
dtypes: int64(2), str(10)
memory usage: 9.7 MB


**컬럼 수 : 25개 → 11개**  
**결측치 : 0**

In [189]:
# 모든 컬럼명을 대상으로 첫 글자 대문자로 변경
new_df.columns = [col.title() for col in new_df.columns]

new_df.head()

Article_Id  Product_Code          Prod_Name Product_Type_Info  \
0   108775015        108775          Strap top    Vest top (253)   
1   108775044        108775          Strap top    Vest top (253)   
2   108775051        108775      Strap top (1)    Vest top (253)   
3   110065001        110065  OP T-shirt (Idro)         Bra (306)   
4   110065002        110065  OP T-shirt (Idro)         Bra (306)   

   Product_Group_Name Graphical_Appearance_Info Perceived_Colour_Master_Info  \
0  Garment Upper body           Solid (1010016)                    Black (5)   
1  Garment Upper body           Solid (1010016)                    White (9)   
2  Garment Upper body          Stripe (1010017)                    White (9)   
3           Underwear           Solid (1010016)                    Black (5)   
4           Underwear           Solid (1010016)                    White (9)   

         Department_Info            Index_Info Index_Group_Info  \
0    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
1    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
2    Jersey Basic (1676)        Ladieswear (A)   Ladieswear (1)   
3  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   
4  Clean Lingerie (1339)  Lingeries/Tights (B)   Ladieswear (1)   

                  Section_Info        Garment_Group_Info  
0  Womens Everyday Basics (16)       Jersey Basic (1002)  
1  Womens Everyday Basics (16)       Jersey Basic (1002)  
2  Womens Everyday Basics (16)       Jersey Basic (1002)  
3         Womens Lingerie (61)  Under-, Nightwear (1017)  
4         Womens Lingerie (61)  Under-, Nightwear (1017)

In [190]:
new_df.shape

(105542, 12)

In [191]:
new_df.to_csv('cleaned_articles_hm.csv', index = False)